shopping/src/main/ui/src/components/Card/Card.js

```diff
@@ -0,0 +1,114 @@
+import React, {PureComponent} from 'react';
+import "./Card.css";
+import {orderShirts} from "../../constants";
+import classnames from "classnames";
+
+export const messageTypes = {
+    SUCCESS: "SUCCESS",
+    FAILURE: "FAILURE",
+    WARNING: "WARNING",
+}
+
+
+class Card extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            value: 1,
+            isLoading: false,
+            alert: null,
+        };
+
+        this.selectRef = React.createRef();
+    }
+
+    getOptions = () => {
+        const options = [];
+        for (let i=1; i<=10; i++) {
+            options.push(<option value={i} key={i}>{i}</option>)
+        }
+        return options;
+    }
+
+    handleChange = () => this.setState({value: this.selectRef.current.value});
+
+    handleOrder = () => {
+        this.setState({isLoading: true, alert: null});
+        const start = Date.now();
+        orderShirts(this.props.item.name, this.state.value)
+        .then((data) => {
+            const end = Date.now();
+            const duration = ((end - start) >= 1000) ? `${(end - start)/1000}s` : `${(end - start)}ms`;
+            this.setState({
+                isLoading: false,
+                alert: {
+                    type: data.status === 200 ? messageTypes.SUCCESS : messageTypes.FAILURE,
+                    header: data.status === 200 ? `Order succeeded after ${duration}` : `Order failed after ${duration}. try again!`,
+                }
+            });
+        })
+    }
+
+    closeAlert = () => this.setState({alert: null});
+
+    render() {
+        const { item } = this.props;
+        const { alert } = this.state;
+        return (
+            <div className="card clr-col-6 clr-col-sm-6 clr-col-md-4 clr-col-lg-3 clr-col-xl-2">
+                <div className="card-header">
+                    {item.name}
+                </div>
+                <div className="card-block image-container">
+                    <img src={item.src} alt="item.name"/>
+                </div>
+                <div className="card-footer">
+                    <div className="clr-row">
+                        <select
```

Comment:
Maybe using some nice looking 3rd-party selection component. We tried multiple devices, seems Android phone having the problem as follows:
<img src="https://user-images.githubusercontent.com/7380836/57414272-4952fd80-71ac-11e9-8337-9392d0381d37.jpg" data-canonical-src="https://user-images.githubusercontent.com/7380836/57414272-4952fd80-71ac-11e9-8337-9392d0381d37.jpg" width="200" height="400" />